backend/populate_catalog/task.py

```diff
@@ -81,9 +81,10 @@ def get_catalog_href(self, cat: Catalog, parent_dir: str, is_root: bool) -> str:
         return join(cat_root, original_path[-1])
 
     def get_collection_href(self, col: Collection, parent_dir: str, is_root: bool) -> str:
-        original_path = urlparse(col.get_self_href()).path.rsplit("/", maxsplit=2)
         assert not is_root
-        return join(parent_dir, *original_path[-2:])
+        dataset_path_prefix = "/".join(parent_dir.split("/")[:4])
```

Comment:
This is ugly I know - but I played around with too many variations and decided i;d just ask :)